dot/peerset/peerset_test.go

```diff
@@ -125,58 +120,60 @@ func TestReAllocAfterBanned(t *testing.T) {
 	}
 
 	// We ban a node by setting its reputation under the threshold.
+	processor.EXPECT().Process(Message{Status: Drop, setID: 0, PeerID: peer1})
 	rep := newReputationChange(BannedThresholdValue-1, "")
+
 	// we need one for the message to be processed.
+	processor.EXPECT().Process(Message{Status: Reject, setID: 0, PeerID: peer1})
 	handler.ReportPeer(rep, peer1)
 	time.Sleep(time.Millisecond * 100)
 
-	checkMessageStatus(t, <-ps.resultMsgCh, Drop)
-
 	// Check that an incoming connection from that node gets refused.
-
+	processor.EXPECT().Process(Message{Status: Connect, setID: 0, PeerID: peer1})
 	handler.Incoming(0, peer1)
-	checkMessageStatus(t, <-ps.resultMsgCh, Reject)
+	time.Sleep(time.Second * 2)
```

Comment:
Why do we need a sleep at the end of the test??

Reply:
Because we need to wait for the `periodicallyAllocSlots` being triggered and reject the connection

Reply:
And this one is called from where? Isn't there a way to wait for it with some event-driven approach?
A sleep is already bad, but at the end of a test this really looks super sketchy :smile: 
Or at least add a comment why we sleep. I guess we wait for all mock calls to be executed so the mock controller doesn't complain right?

Reply:
The `periodicallyAllocSlots` is started in the `newTestPeerSet` helper function when we call `handler.Start()`